src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -206,20 +237,29 @@ public void setTags(Set<Tag> tags) {
         }
 
         /**
-         * Sets {@code notes} to this object's {@code notes}.
-         * A defensive copy of {@code notes} is used internally.
+         * Returns an unmodifiable note list, which throws {@code UnsupportedOperationException}
+         * if modification is attempted.
+         * Returns {@code Optional#empty()} if {@code strengths} is null.
          */
-        public void setNotes(List<Note> notes) {
-            this.notes = (notes != null) ? new ArrayList<>(notes) : null;
+        public Optional<List<Note>> getStrengths() {
+            return (miscellaneous != null) ? Optional.of(Collections.unmodifiableList(strengths)) : Optional.empty();
+        }
+        /**
+         * Returns an unmodifiable note list, which throws {@code UnsupportedOperationException}
+         * if modification is attempted.
+         * Returns {@code Optional#empty()} if {@code weaknesses} is null.
+         */
+        public Optional<List<Note>> getWeaknesses() {
+            return (miscellaneous != null) ? Optional.of(Collections.unmodifiableList(weaknesses)) : Optional.empty();
```

Comment:
Not sure if this should be
```suggestion
            return (weaknesses != null) ? Optional.of(Collections.unmodifiableList(weaknesses)) : Optional.empty();
```